src/validation/tests/property_validation_tests.rs

```diff
@@ -107,3 +109,44 @@ fn property_with_var_output_in_get_block() {
       |
     ");
 }
+
```

Comment:
Some more tests I would like to see:
* VAR foo .. defined in parent, property in child, this should be a duplicate
* Property with getter defined in parent, setter in child. Setting the value in the parent should be an error, but not in the child.
* property defined in parent, variable in child should be an error

Reply:
I think extension and overrides of properties are covered in separate PRs, @volsa please correct me if I'm wrong

Reply:
Alright, I've addressed the first and third bulletpoint with the latest commit; could you do a second round of review @ghaith ?

Also can you elaborate on the second bullet point, not sure I totally understood what you mean here?

Reply:
Defining a  setter in the child instead of the parent,  means the parent will have an error when writing to the child, but the child can write and read the parent property fine.
```iecst
FUNCTION_BLOCK parent
PROPERTY myProp : DINT
GET END_GET
END_PROPERTY
myProp := 5; //Error
myProp; //OK
END_FUNCTION_BLOCK

FUNCTION_BLOCK child EXTENDS parent
PROPERTY myProp : DINT
SET END_SET
END_PROPERTY
myProp := 5; //OK
myProp := myProp  + 1; //OK
END_FUNCTION_BLOCK

FUNCTION main : DINT
VAR
  parent_fb: 
  child_fb: child;
END_VAR
parent_fb.myProp := 5; //Error
child_fb.myProp := 5; //OK
child_fb.myProp := parent_fb.myProp; //OK
child_fb.myProp := child_fb.myProp + 1; //OK
END_FUNCTION
```